### ISSUE TITLE
TS2454 - Variable is used before being assigned. (Variable is assigned in an always-true condition)

### ISSUE THREAD

**Issue by thegreatzeus** [microsoft/TypeScript#54004]
# Bug Report

### 🔎 Search Terms

1. Variable 'x' is used before being assigned.
2. TS2454 - Variable is used before being assigned error
3. var causing TS2454 when flow explicitly sets the value of the var

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about TS2454 - Variable is used before being assigned
- The same behavior exists in the Nightly version

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=5.0.4#code/DYUwLgBAxg9gdgEwJZifCBeCYBOBXEAbgFgAoMpAMwgApZEU04BKCAbzLIm4gDcBDHBAAembPiJkAvp1L0AzjFAA6YDADmNYc0JA)

### 💻 Code

```ts
let condition = true;

if (condition) {

    var x = true;
}

console.log(x);
```

### 🙁 Actual behavior

TypeScript throws 'TS2454' "Variable 'x' is used before being assigned."

### 🙂 Expected behavior

No errors should be thrown since the flow explicitly assigns a value to variable 'x'

**Comment by RyanCavanaugh** [comment]
Either `condition` is *actually* always true, in which case this code is at best in a transient state, or it's *not* actually always true, in which case the error is correct. Or maybe it really is always true and you can move the `console.log` inside the block for clarity.

**Comment by thegreatzeus** [comment]
@RyanCavanaugh I appreciate your timely response.

I'm having some difficulty understanding the first part of your answer, could you elaborate on what you meant by the code being in _a transient state_? Thank you so much.

**Comment by fatcerberus** [comment]
@thegreatzeus In other words it may be always-true *now*, but it’s not likely to stay that way—otherwise you wouldn’t have written the `if`.

**Comment by RyanCavanaugh** [comment]
Yeah, I use "transient" to mean "a person is in the middle of editing this code" and their likely next action is to either remove the `if () {` blocking, or change `condition`'s initializer to something dynamic.

**Comment by thegreatzeus** [comment]
Thanks, @fatcerberus and @RyanCavanaugh. I understand the logic. 
Have a good rest of your day!
